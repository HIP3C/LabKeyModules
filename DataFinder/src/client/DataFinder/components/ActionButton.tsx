import * as React from 'react';
import { GroupInfo } from '../../typings/CubeData';

interface ActionButtonProps {
    onClick: () => void;
    text: string;
}

export const ActionButton: React.FC<ActionButtonProps> = (props) => {
    return (
        <button onClick={props.onClick} disabled={false}>
            {props.text}
        </button>

    )
}

interface LoadDropdownProps {
    groups: GroupInfo[],
    loadParticipantGroup: (groupInfo: GroupInfo) => void
}
export const LoadDropdown: React.FC<LoadDropdownProps> = ({ groups, loadParticipantGroup }) => {
    const buttonData = groups.map((group) => {
        return ({
            label: group.label,
            action: () => loadParticipantGroup(group),
            disabled: false
        })
    })
    return (
        <DropdownButtons buttonData={buttonData} title={"Load"} />
    )
}

interface SaveDropdownProps {
    saveAs: () => void;
    save: () => void;
    disableSave: boolean
}

export const SaveDropdown: React.FC<SaveDropdownProps> = ({ saveAs, save, disableSave }) => {
    const buttonData = [
        {
            label: "Save",
            action: save,
            disabled: disableSave
        },
        {
            label: "Save As",
            action: saveAs,
            disabled: false
        }
    ]
    return (
        <DropdownButtons title="Save" buttonData={buttonData} />
    )
}

interface ClearDropdownProps {
    clearAll: () => void;
    reset: () => void;
}

export const ClearDropdown: React.FC<ClearDropdownProps> = ({ clearAll, reset }) => {
    const buttonData = [
        {
            label: "Clear All",
            action: clearAll,
            disabled: false
        },
        {
            label: "Clear Unsaved Changes",
            action: reset,
            disabled: false
        }
    ]
    return (
        <DropdownButtons title={"Clear"} buttonData={buttonData} />
    )
}




interface DropdownButtonProps {
    title: string
    buttonData: {
        label: string;
        action: () => void;
        disabled: boolean
    }[]
}
export const DropdownButtons: React.FC<DropdownButtonProps> = ({ title, buttonData }) => {
    return (
        <div className="dropdown" style={{  display: "inline-block", margin: "5px" }}>
            <div className="btn df-dropdown-button" role="group" >
                <button className="btn btn-default dropdown-toggle" type="button" id={"button-" + title} data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
                    <span>{title}</span>
                    <span style={{paddingLeft:"5px"}}><i className="fa fa-caret-down"></i></span>
                </button>
                <ul className="dropdown-menu filter-dropdown" aria-labelledby={"button-" + title}>
                    {buttonData.map((button) => {
                        return (
                            <li className={button.disabled ? "disabled" : ""}>
                                <a key={button.label} onClick={button.action} href="#">
                                    {button.label}
                                </a>
                            </li>
                        )
                    })}
                </ul>

            </div>
        </div>
    )
}
