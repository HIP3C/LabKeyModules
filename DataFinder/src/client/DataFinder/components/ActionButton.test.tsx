import React from 'react'
import * as renderer from 'react-test-renderer'
import * as ab from './ActionButton'
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

describe("<ActionButton />", () => {
    test("Apply", () => {

        const tree = renderer.create(<ab.ActionButton onClick={() => { }} text={"Apply"} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe("<DropdownButtons />", () => {
    test("Save", () => {
        const buttonData = [
            {
                label: "Save",
                action: () => { },
                disabled: false
            },
            {
                label: "Save As",
                action: () => {},
                disabled: true
            }
        ]
        const tree = renderer.create(<ab.DropdownButtons title={"Save"} buttonData={buttonData} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe("<ClearDropdown/>", () => {
    test("Clear", () => {
        const tree = renderer.create(<ab.ClearDropdown clearAll={() => {}} reset={() => {}} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe("<SaveDropdown />", () => {
    test("enabled", () => {
        const tree = renderer.create(<ab.SaveDropdown saveAs={() => {}} save={() => {}} disableSave={false} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test("disabled", () => {
        const tree = renderer.create(<ab.SaveDropdown saveAs={() => {}} save={() => {}} disableSave={true} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    test("enabled to disabled", () => {
        const button = mount(<ab.SaveDropdown saveAs={jest.fn()} save={jest.fn()} disableSave={false}/>);

        // Ensure there are no disabled li tags
        expect(button.find('li.disabled').length).toEqual(0);

        // Update props and verify
        button.setProps({disableSave: true});

        // Ensure there is now one disabled li tags
        expect(button.find('li.disabled').length).toEqual(1)

        expect(toJson(button)).toMatchSnapshot();
        button.unmount();
    })
})

