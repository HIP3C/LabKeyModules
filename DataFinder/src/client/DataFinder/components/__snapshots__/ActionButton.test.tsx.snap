// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton /> Apply 1`] = `
<button
  disabled={false}
  id="action-button-Apply"
  onClick={[Function]}
>
  Apply
</button>
`;

exports[`<ClearDropdown/> Clear 1`] = `
<div
  className="dropdown"
  style={
    Object {
      "display": "inline-block",
      "margin": "5px",
    }
  }
>
  <div
    className="btn df-dropdown-button"
    role="group"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      className="btn btn-default dropdown-toggle"
      data-toggle="dropdown"
      id="Clear-participant-group-btn"
      type="button"
    >
      <span>
        Clear
      </span>
      <span
        style={
          Object {
            "paddingLeft": "5px",
          }
        }
      >
        <i
          className="fa fa-caret-down"
        />
      </span>
    </button>
    <ul
      aria-labelledby="button-Clear"
      className="dropdown-menu filter-dropdown"
    >
      <li
        className=""
      >
        <a
          href="#"
          id="Clear-All-link"
          onClick={[Function]}
        >
          Clear All
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#"
          id="Clear-Unsaved Changes-link"
          onClick={[Function]}
        >
          Clear Unsaved Changes
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<DropdownButtons /> Save 1`] = `
<div
  className="dropdown"
  style={
    Object {
      "display": "inline-block",
      "margin": "5px",
    }
  }
>
  <div
    className="btn df-dropdown-button"
    role="group"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      className="btn btn-default dropdown-toggle"
      data-toggle="dropdown"
      id="Save-participant-group-btn"
      type="button"
    >
      <span>
        Save
      </span>
      <span
        style={
          Object {
            "paddingLeft": "5px",
          }
        }
      >
        <i
          className="fa fa-caret-down"
        />
      </span>
    </button>
    <ul
      aria-labelledby="button-Save"
      className="dropdown-menu filter-dropdown"
    >
      <li
        className=""
      >
        <a
          href="#"
          id="Save-link"
          onClick={[Function]}
        >
          Save
        </a>
      </li>
      <li
        className="disabled"
      >
        <a
          href="#"
          id="Save-As-link"
          onClick={[Function]}
        >
          Save As
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<SaveDropdown /> disabled 1`] = `
<div
  className="dropdown"
  style={
    Object {
      "display": "inline-block",
      "margin": "5px",
    }
  }
>
  <div
    className="btn df-dropdown-button"
    role="group"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      className="btn btn-default dropdown-toggle"
      data-toggle="dropdown"
      id="Save-participant-group-btn"
      type="button"
    >
      <span>
        Save
      </span>
      <span
        style={
          Object {
            "paddingLeft": "5px",
          }
        }
      >
        <i
          className="fa fa-caret-down"
        />
      </span>
    </button>
    <ul
      aria-labelledby="button-Save"
      className="dropdown-menu filter-dropdown"
    >
      <li
        className="disabled"
      >
        <a
          href="#"
          id="Save-link"
          onClick={[Function]}
        >
          Save
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#"
          id="Save-As-link"
          onClick={[Function]}
        >
          Save As
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<SaveDropdown /> enabled 1`] = `
<div
  className="dropdown"
  style={
    Object {
      "display": "inline-block",
      "margin": "5px",
    }
  }
>
  <div
    className="btn df-dropdown-button"
    role="group"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      className="btn btn-default dropdown-toggle"
      data-toggle="dropdown"
      id="Save-participant-group-btn"
      type="button"
    >
      <span>
        Save
      </span>
      <span
        style={
          Object {
            "paddingLeft": "5px",
          }
        }
      >
        <i
          className="fa fa-caret-down"
        />
      </span>
    </button>
    <ul
      aria-labelledby="button-Save"
      className="dropdown-menu filter-dropdown"
    >
      <li
        className=""
      >
        <a
          href="#"
          id="Save-link"
          onClick={[Function]}
        >
          Save
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#"
          id="Save-As-link"
          onClick={[Function]}
        >
          Save As
        </a>
      </li>
    </ul>
  </div>
</div>
`;
